### ISSUE TITLE
An error occurred while Style/GuardClause was inspecting `rescue` statement

### ISSUE THREAD

**Issue by btrautmann** [rubocop/rubocop#11533]
## Expected behavior

Rubocop should analyze the ruby code without errors unrelated to the code itself.

## Actual behavior

Rubocop fails due to `GuardClause` cop:

`» bundle exec rubocop . --debug --config=.rubocop.yml`
```console
configuration from /Users/brandontrautmann/Development/mobile/flutter/.rubocop.yml
Default configuration from /Users/brandontrautmann/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/rubocop-1.40.0/config/default.yml
...
14 files inspected, no offenses detected

1 error occurred:
An error occurred while Style/GuardClause cop was inspecting /Users/brandontrautmann/Development/mobile/flutter/app/ios/fastlane/Fastfile:124:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.0.3) [arm64-darwin21]
Finished in 16.153554999968037 seconds
```

## Steps to reproduce the problem

The offending code seems to be:

```ruby
  rescue StandardError => e
    if e.message.include? 'Another build is in review'
      exit(0)
    else
      raise
    end
  end
```

## RuboCop version

```console
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.0.3) [arm64-darwin21]
```

**Comment by koic** [comment]
This issue has been resolved in the latest RuboCop (1.44.1) . Can you upgrade RuboCop to the latest version?

**Comment by btrautmann** [comment]
Yes, that worked! Thanks @koic!
